### ISSUE TITLE
Bug in CrossPlatLock?

### ISSUE THREAD

**Issue by cheenamalhotra** [AzureAD/microsoft-authentication-extensions-for-dotnet#204]
Noticed this in CrossPlatLock constructor:

```
      catch (IOException ex)
      {
          fileStream?.Dispose();
          fileStream = null;
          exception = ex;
          Thread.Sleep(lockFileRetryDelay);
      }
      catch (UnauthorizedAccessException ex)
      {
          fileStream?.Dispose();
          fileStream = null;
          exception = ex;
          Thread.Sleep(lockFileRetryCount);
      }
```

You probably want to sleep for `lockFileRetryDelay` instead of `lockFileRetryCount` in the second catch block.

**Comment by bgavrilMS** [comment]
Absolutely

**Comment by bgavrilMS** [comment]
Will be fixed with the next version of MSAL - 4.51.0 and the associated extension.
